### DISCUSSION TITLE
Where are stored the docker images that's running on each tasks?

### DISCUSSION THREAD

**Discussion by akaibukai** [concourse/concourse#8931] [Help & Support]
Hi there..
I'm trying Concourse and it seems it's always pulling images when a job is running..
I wanted to know where the images are stored?
After (and even during) running a job the image the job should be running within is not listed by my `docker images` command..
Also, is there a way to have the images kept in the host so that it's not pulling the image each time?
I hope this question makes sense and it's the correct place to ask.

**Comment by taylorsilva** [comment]
Images on a Concourse worker are not stored in any local registry that Docker would know to read about. Images are downloaded and converted into what Concourse calls `volumes`. You can see `volumes` on the diagram here: https://concourse-ci.org/internals.html
![concourse_architecture](https://github.com/concourse/concourse/assets/4583837/0787c19e-99a0-4dc0-9e3e-3126f62ff6ee)

> Also, is there a way to have the images kept in the host so that it's not pulling the image each time?

Resources within Concourse are cached for a period of time. I forget how long and if that's configurable or not. So if you run the same job twice, with the second time being shortly after the first, you'll notice all your `get` steps finish fast because they were in the cache.

If you want some images always available, I've had this idea of hacking on the steps here: https://concourse-ci.org/concourse-worker.html#installing-or-upgrading-bundled-resource-types
with the `mirror_self` function of the `mock-resource` type: https://github.com/concourse/mock-resource

Requires extra steps on an operators part to set that up of course.
